From d8b61898fc7eec006f5d1ea84249aa39bae79a1b Mon Sep 17 00:00:00 2001
From: Yangbo Lu <yangbo.lu@nxp.com>
Date: Wed, 5 Dec 2018 13:28:40 +0800
Subject: [PATCH 383/706] arm64: dts: ls1028a: support SD UHS-I and eMMC HS200

This patch is to add properties for SD UHS-I and
eMMC HS200 for ls1028ardb board.

Signed-off-by: Yangbo Lu <yangbo.lu@nxp.com>
(cherry picked from commit 3d17aae8480efd35d05dab954a308df5e9c14890)
Signed-off-by: Zumeng Chen <zumeng.chen@windriver.com>
---
 arch/arm64/boot/dts/freescale/fsl-ls1028a-qds.dts |  8 ++++++++
 arch/arm64/boot/dts/freescale/fsl-ls1028a-rdb.dts | 13 +++++++++++++
 arch/arm64/boot/dts/freescale/fsl-ls1028a.dtsi    |  4 ++++
 3 files changed, 25 insertions(+)

diff --git a/arch/arm64/boot/dts/freescale/fsl-ls1028a-qds.dts b/arch/arm64/boot/dts/freescale/fsl-ls1028a-qds.dts
index c5b1ec72a7a5..41082d221ded 100755
--- a/arch/arm64/boot/dts/freescale/fsl-ls1028a-qds.dts
+++ b/arch/arm64/boot/dts/freescale/fsl-ls1028a-qds.dts
@@ -108,6 +108,14 @@
 	};
 };
 
+&esdhc {
+	status = "okay";
+};
+
+&esdhc1 {
+	status = "okay";
+};
+
 &dp0 {
 	status = "okay";
 };
diff --git a/arch/arm64/boot/dts/freescale/fsl-ls1028a-rdb.dts b/arch/arm64/boot/dts/freescale/fsl-ls1028a-rdb.dts
index 4f4d1ff73e1a..104beb00d735 100755
--- a/arch/arm64/boot/dts/freescale/fsl-ls1028a-rdb.dts
+++ b/arch/arm64/boot/dts/freescale/fsl-ls1028a-rdb.dts
@@ -106,6 +106,19 @@
 	};
 };
 
+&esdhc {
+	status = "okay";
+	sd-uhs-sdr104;
+	sd-uhs-sdr50;
+	sd-uhs-sdr25;
+	sd-uhs-sdr12;
+};
+
+&esdhc1 {
+	status = "okay";
+	mmc-hs200-1_8v;
+};
+
 &i2c0 {
 	status = "okay";
 	pca9547@77 {
diff --git a/arch/arm64/boot/dts/freescale/fsl-ls1028a.dtsi b/arch/arm64/boot/dts/freescale/fsl-ls1028a.dtsi
index 1b3d9b63d0e5..1773594b4793 100755
--- a/arch/arm64/boot/dts/freescale/fsl-ls1028a.dtsi
+++ b/arch/arm64/boot/dts/freescale/fsl-ls1028a.dtsi
@@ -516,10 +516,12 @@
 			reg = <0x0 0x2140000 0x0 0x10000>;
 			interrupts = <0 28 0x4>; /* Level high type */
 			clock-frequency = <0>;
+			clocks = <&clockgen 2 1>;
 			voltage-ranges = <1800 1800 3300 3300>;
 			sdhci,auto-cmd12;
 			little-endian;
 			bus-width = <4>;
+			status = "disabled";
 		};
 
 		esdhc1: esdhc@2150000 {
@@ -527,11 +529,13 @@
 			reg = <0x0 0x2150000 0x0 0x10000>;
 			interrupts = <0 63 0x4>; /* Level high type */
 			clock-frequency = <0>;
+			clocks = <&clockgen 2 1>;
 			voltage-ranges = <1800 1800 3300 3300>;
 			sdhci,auto-cmd12;
 			broken-cd;
 			little-endian;
 			bus-width = <4>;
+			status = "disabled";
 		};
 
 		sata: sata@3200000 {
-- 
2.17.1

